sgtm/dioscuri/dioscuri_get_gemini_response.tpl

```diff
@@ -0,0 +1,653 @@
+        "simpleValueType": true,
+        "defaultValue": 1,
+        "help": "Positive values penalize tokens that repeatedly appear in the generated text, decreasing the probability of repeating content.  This maximum value for frequencyPenalty is up to, but not including, 2.0. Its minimum value is -2.0.",
+        "valueValidators": [
+          {
+            "type": "NON_EMPTY"
+          }
+        ]
+      },
+      {
+        "type": "TEXT",
+        "name": "presencePenalty",
+        "displayName": "Presence Penalty",
+        "simpleValueType": true,
+        "defaultValue": 1,
+        "help": "Positive values penalize tokens that already appear in the generated text, increasing the probability of generating more diverse content. This maximum value for presencePenalty is up to, but not including, 2.0. Its minimum value is -2.0.",
+        "valueValidators": [
+          {
+            "type": "NON_EMPTY"
+          }
+        ],
+        "enablingConditions": [
+          {
+            "paramName": "geminiModel",
+            "paramValue": "gemini-1.5-pro",
+            "type": "EQUALS"
+          }
+        ]
+      },
+      {
+        "type": "TEXT",
+        "name": "stopSequences",
+        "displayName": "Stop Sequences",
+        "simpleValueType": true,
+        "help": "Specifies a list of strings that tells the model to stop generating text if one of the strings is encountered in the response. If a string appears multiple times in the response, then the response truncates where it\u0027s first encountered. The strings are case-sensitive.",
+        "valueValidators": [
+          {
+            "type": "REGEX",
+            "args": [
+              "[a-z]+(?:, +?[a-z]+){0,4}"
+            ],
+            "errorMessage": "Must be a comma-separated list containing a maximum of five words"
+          }
+        ]
+      }
+    ]
+  }
+]
+
+
+___SANDBOXED_JS_FOR_SERVER___
+
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/**
+ * @fileoverview sGTM variable tag that makes a call to Gemini API
+ * @version 1.0.0
+ */
+const getGoogleAuth = require("getGoogleAuth");
+const JSON = require("JSON");
+const logToConsole = require("logToConsole");
+const sendHttpRequest = require("sendHttpRequest");
+
+//Function to convert string of words to array of strings
+function splitAndTrim(commaSeparatedString) {
+  commaSeparatedString = "hello, goodbye";
```

Comment:
Should this line be here? It's overwriting the argument

Reply:
Was used for testing - removed